core/src/main/java/org/openconnectors/connect/SinkConnector.java

```diff
@@ -28,15 +28,15 @@
  *
  * @param <U> Type for messages to be submitted for publishing
  */
-public interface SinkConnector<U> extends Connector {
+public interface SinkConnector<U, R> extends Connector {
 
     /**
      * Attempt to publish a type safe collection of messages
      *
      * @param messages Objects to process in the sink
      * @return Completable future fo async publish request
      */
-    CompletableFuture<Void> publish(final Collection<U> messages);
+    CompletableFuture<R> publish(final Collection<U> messages);
```

Comment:
If we don't need a value in the connector framework, we should keep it to `Void`. 

Then, in the Pulsar sink we can easily transform a `CompletableFuture<MessageId>` into a `CompletableFuture<Void>` by using a `.thenApply(x -> null)` mapping function.

Reply:
agree with @merlimat , changing void make the interfaces more complicated then need be. The focus of the framework has to be simplicity.

Reply:
ok

Reply:
I have changed the return type back to CompletableFuture<Void>

Reply:
but you are still taking a parameter for it in the generic interface, it should be removed

Reply:
sorry, I miss that, I have removed them now